src/strategyHelpers/blueGreen/smiBlueGreenHelper.ts

```diff
@@ -1,163 +1,102 @@
-   const workloadDeployment = await createWorkloadsWithLabel(
-      kubectl,
-      manifestObjects.deploymentEntityList,
-      GREEN_LABEL_VALUE
-   )
-
-   return {workloadDeployment, newObjectsList}
-}
-
-export async function promoteBlueGreenSMI(kubectl: Kubectl, manifestObjects) {
-   // checking if there is something to promote
-   if (
-      !(await validateTrafficSplitsState(
-         kubectl,
-         manifestObjects.serviceEntityList
-      ))
-   ) {
-      throw Error('Not in promote state SMI')
-   }
-
-   // create stable deployments with new configuration
-   return await createWorkloadsWithLabel(
-      kubectl,
-      manifestObjects.deploymentEntityList,
-      NONE_LABEL_VALUE
-   )
-}
-
-export async function rejectBlueGreenSMI(
-   kubectl: Kubectl,
-   filePaths: string[],
-   annotations: {[key: string]: string} = {}
-) {
-   // get all kubernetes objects defined in manifest files
-   const manifestObjects: BlueGreenManifests = getManifestObjects(filePaths)
-
-   // route trafficsplit to stable deployments
-   await routeBlueGreenSMI(
-      kubectl,
-      NONE_LABEL_VALUE,
-      manifestObjects.serviceEntityList,
-      annotations
-   )
-
-   // delete rejected new bluegreen deployments
-   await deleteWorkloadsWithLabel(
-      kubectl,
-      GREEN_LABEL_VALUE,
-      manifestObjects.deploymentEntityList
-   )
-
-   // delete trafficsplit and extra services
-   await cleanupSMI(kubectl, manifestObjects.serviceEntityList)
-}
+import {
+   K8sDeleteObject,
+   K8sObject,
+   TrafficSplitObject
+} from '../../types/k8sObject'
+import {DeployResult} from '../../types/deployResult'
+import {getInputAnnotations} from '../../utilities/inputUtils'
+
+export const TRAFFIC_SPLIT_OBJECT_NAME_SUFFIX = '-trafficsplit'
+export const TRAFFIC_SPLIT_OBJECT = 'TrafficSplit'
+export const MIN_VAL = 0
+export const MAX_VAL = 100
 
 export async function setupSMI(
    kubectl: Kubectl,
-   serviceEntityList: any[],
-   annotations: {[key: string]: string} = {}
-) {
+   serviceEntityList: any[]
+): Promise<BlueGreenDeployment> {
    const newObjectsList = []
    const trafficObjectList = []
 
    serviceEntityList.forEach((serviceObject) => {
       // create a trafficsplit for service
       trafficObjectList.push(serviceObject)
       // set up the services for trafficsplit
-      const newStableService = getSMIServiceResource(
-         serviceObject,
-         STABLE_SUFFIX
-      )
-      const newGreenService = getSMIServiceResource(serviceObject, GREEN_SUFFIX)
+      const newStableService = getStableSMIServiceResource(serviceObject)
+      const newGreenService = getGreenSMIServiceResource(serviceObject)
       newObjectsList.push(newStableService)
       newObjectsList.push(newGreenService)
    })
 
    // create services
-   const manifestFiles = fileHelper.writeObjectsToFile(newObjectsList)
-   await kubectl.apply(manifestFiles)
+   let servicesDeploymentResult: DeployResult = await deployObjects(
```

Comment:
this should be const too. just use const by default unless you absolutely need to use let

Reply:
will replace lets with const wherever I can